### DISCUSSION TITLE
How GO-CAM links pathways, enzymatic reactions, and metabolites to GO terms?

### DISCUSSION THREAD

**Discussion by 0m1n0** [geneontology/helpdesk#397] [Q&A] [answered]
Dear GO team,

First of all, I would like to thank all your colleagues who are doing a great job. I work on metabolic pathways by diving into the world of ontology. That's why I'm very interested in GO-CAM, and I found the idea brilliant!

The [ttl file](http://geneontology.org/docs/download-go-cams/) downloadable from the official website does not contain details on reactions and metabolites. And I saw on the FAQ that 

> The GO-CAM model is simpler than that used by databases such as Reactome: in a GO-CAM we do not typically capture all the participants in a reaction, together with their stoichiometry. Instead this information is included in the GO term.

My understanding from reading the papers and documentation in your GitHub repositories is that you aim to integrate metabolic pathways such as Reactome using [Pathways2GO](https://github.com/geneontology/pathways2GOl) and complete catalytic reactions (enzymes, metabolites ...etc) using on Rhea.

For example, the [reactome-go-cams](https://github.com/geneontology/reactome-go-cams)  shows how to build GO-CAM from Reactome pathways. But the [ttl files](https://github.com/geneontology/reactome-go-cams/tree/master/models) do not reflect 100% of what is observed on Noctua. 

For Rhea, I found [go-rhea-boom](https://github.com/geneontology/go-rhea-boom) but I have the impression that the json files do not contain the metabolites (the biochemical equation is annotated in the label).

### Considering all these, here are my questions:

1. How GO-CAM links pathways, enzymatic reactions, and metabolites to GO terms? What would be the data structure (e.g. what RDF/OWL vocabularies are used)?
2. Do you have a diagram of RDF/OWL overview such as [ChEMBL documentation](https://www.ebi.ac.uk/rdf/documentation/chembl/)?
3. According to the paper ([Thomas et al., 2019](https://www.nature.com/articles/s41588-019-0500-1)) GO-CAM models use the Evidence and Conclusion Ontology (ECO) but it seems empty (see [Ontobee](https://www.ontobee.org/ontology/ECO)). Why?
4. Why do SPARQL queries on the downloaded ttl file not work as expected?

For the last question, I ran some queries described on Pathways2GO. For example [get_chemicals.rq](https://github.com/geneontology/pathways2GO/blob/master/exchange/src/main/resources/org/geneontology/gocam/exchange/get_chemicals.rq): this query works on [Gene Ontology SPARQL endpoint](http://geneontology.org/sparql) but not locally. I guess it' s because the ttl file only contains GO-CAM models?

I apologize for overwhelming you with so many questions. 
Your answers will greatly help me. 

Regards,
Min

**Comment by balhoff** [comment, marked as answer]
@0m1n0 sorry for the slow response.

1. In order to completely interpret a GO-CAM OWL file, you must import the ontologies from which the classes mentioned in the GO-CAM come from. We publish a large merged ontology which provides the vocabulary used within GO-CAMs: `http://purl.obolibrary.org/obo/go/extensions/go-lego.owl`
   
   This ontology links many GO terms to chemical inputs and outputs, as well as anatomical structures, etc.

2. I don't think we have a great diagram available like that. We do have a schema for GO-CAMs in the [ShEx](http://shex.io) language here: https://github.com/geneontology/go-shapes/blob/master/shapes/go-cam-shapes.shex

3. I can't check Ontobee right now, since there seems to be a problem with their server. But ECO should not be empty (it's imported into go-lego). You can download it here: `http://purl.obolibrary.org/obo/eco.owl`

4. To do SPARQL queries using the ontology terms you will need to merge the GO-CAM with the go-lego ontology. Our SPARQL endpoint also includes some precomputed OWL inferences which will affect which results you get from a SPARQL query.

**Reply by 0m1n0** [reply]
@balhoff thank you for your detailed explanations and the various useful links.
ShEx language is sufficiently descriptive to understand GO-CAM structure.
After downloading both OWL and ttl files, I tried to reproduce SPARQL queries which are shown on [GO-CAM's SPARQL doc](http://geneontology.org/docs/sparql) but it doesn't work. Even pre-selection by `GRAPH <model_url>` doesn't work. I will continue to dig through the data.

**Reply by balhoff** [reply]
There is some processing that happens while the models are imported into the triplestore. We don't have much in the way of documentation on the triplestore so far, as its organization remains a bit in flux. Our tool that loads the models places each one in a graph matching its ontology IRI. That tool also computes OWL inferences based on the go-lego ontology and inserts the inferred triples in a graph matching the ontology IRI with `#inferred` appended. The triplestore loading procedure is defined here: https://github.com/geneontology/go-site/blob/fa72074a9854144a9070b6ed2412144e2f2bd660/pipeline/Makefile#L198-L212 I would be happy provide any other guidance on doing OWL inference on GO-CAMs that would be useful. We have also been talking about other ways to render GO-CAMs that would be useful to the community, such that they wouldn't need OWL expertise to make use of them—ideas are welcome!
